### ISSUE TITLE
Export changeset to file

### ISSUE THREAD

**Issue by soderlind** [xwp/wp-customize-snapshots#150]
Do you plan to add the option to save a changeset to file (json / yml) ?

**Comment by jeffpaul** [comment]
@soderlind interesting request. What's the intent with a JSON/YML output file? To then load into a different WP instance? In which case, we'd need the ability to import as well it seems.

**Comment by soderlind** [comment]
Sure, import, into another site, is given :)

**Comment by westonruter** [comment]
It's a good idea and one that I've thought about. It is trivial to allow for exporting. The challenge comes with _importing_. In particular, conflict resolution will be very challenging when importing a changeset on a non-fresh site since any settings that contain IDs, like nav menus, nav menu items, and widgets instances will all be likely invalid when the changeset is imported.

**Comment by soderlind** [comment]
A start could be allowing import only for fresh sites.

**Comment by westonruter** [comment]
Yes, in fact we have the `fresh_site` option which could be used for this purpose.

**Comment by soderlind** [comment]
Yup, would be like having a tailored starter changeset.

**Comment by westonruter** [comment]
Yeah, and that's something we've talked about before in regards to starter content. A theme could be bundled with multiple starter content sets that the user could choose from, instead of only having one set of starter content. The difference between starter content and changesets is that the former is able to apply to an existing site since it will dynamically compute the IDs and create new objects, whereas the latter (changesets) have static setting IDs and wouldn't apply cleanly to an existing site. But yeah, they could apply to a completely `fresh_site`.
